R/tags.R

```diff
@@ -1947,8 +1947,16 @@ css <- function(..., collapse_ = "") {
     return(NULL)
   }
 
-  # Replace all '.' and '_' in property names to '-'
-  names(props) <- gsub("[._]", "-", tolower(gsub("([A-Z])", "-\\1", names(props))))
+  # Translate camelCase, snake_case, and dot.case to kebab-case
+  # For standard CSS properties only, not CSS variables
+  is_css_var <- grepl("^--", names(props))
+
+  props_std <- names(props)[!is_css_var]
+  props_std <- gsub("([A-Z])", "-\\1", props_std)
+  props_std <- tolower(props_std)
+  props_std <- gsub("[._]", "-", props_std)
+
+  names(props)[!is_css_var] <- props_std
```

Comment:
I'd find this slightly easier to parse mentally if we wrapped the main logic in a function, say something like:

```r
clean_property_names <- function(x) {
  gsub("[._]", "-", tolower(gsub("([A-Z])", "-\\1", names(props))))
}
```

and then use it like:

```r
names(props)[!is_css_var] <- clean_property_names(
  names(props)[!is_css_var]
)
```

Reply:
Sure, although I also find it much easier to parse mentally if we do each step one at a time. So I met in the middle: 3af7643b67e6ffa662d5c4942802f51276aaf1f7